src/chunk.c

```diff
@@ -1156,27 +1157,15 @@ ts_chunk_create_only_table(Hypertable *ht, Hypercube *cube, const char *schema_n
 	return chunk;
 }
 
-#if PG14_GE
-#define OSM_CHUNK_INSERT_CHECK_HOOK "osm_chunk_insert_check_hook"
-typedef int (*ts_osm_chunk_insert_hook_type)(Oid ht_oid, int64 range_start, int64 range_end);
-static ts_osm_chunk_insert_hook_type
-get_osm_chunk_insert_hook()
-{
-	ts_osm_chunk_insert_hook_type *func_ptr =
-		(ts_osm_chunk_insert_hook_type *) find_rendezvous_variable(OSM_CHUNK_INSERT_CHECK_HOOK);
-	return *func_ptr;
-}
-#endif
-
 static Chunk *
 chunk_create_from_hypercube_after_lock(const Hypertable *ht, Hypercube *cube,
 									   const char *schema_name, const char *table_name,
 									   const char *prefix)
 {
 #if PG14_GE
-	ts_osm_chunk_insert_hook_type insert_func_ptr = get_osm_chunk_insert_hook();
+	OsmCallbacks *callbacks = ts_get_osm_callbacks();
 
-	if (insert_func_ptr)
+	if (callbacks)
```

Comment:
I feel that calling this `callbacks` while there might be only one registered handler is a bit misleading - but in the current concept it contains "2" handlers...but since those are both from the same source I wouldn't add the `s`....

Reply:
I can rename it into `osm_callbacks`. Would make more sense?

Reply:
We're getting a struct containing multiple callbacks, we only use one in this particular situation.